server/src/oauth.ts

```diff
@@ -0,0 +1,83 @@
+import OAuth2Strategy from 'passport-oauth2'
+import axios from 'axios'
+import { Role, User } from './entity/User'
+import env from 'dotenv-safe'
+import path from 'path'
+
+env.config({
+  path: path.resolve(process.cwd(), '..', '.env'),
+  example: path.resolve(process.cwd(), '..', '.env.example')
+})
+
+if (process.env.CLIENT_ID === undefined || process.env.CLIENT_SECRET === undefined) {
+  throw new Error('CLIENT_ID AND/OR CLIENT_SECRET ARE NOT SET')
+}
+
+export const oauthStrategy = new OAuth2Strategy({
```

Comment:
Können wir diese ganzen Strings in die .env packen? Interne URLs können auch in die constants Datei, und sollten dann entsprechend überall von dort benutzt werden.